evalml/automl/automl_search.py

```diff
@@ -848,14 +848,18 @@ def close_engine(self):
 
     def _catch_warnings(self, warning_list):
         if len(warning_list) == len(self.allowed_pipelines) and len(warning_list) > 0:
-            # we find the value(s) that we must throw the warning for
+            # we find the value(s) that we must throw any ParameterNotUsedWarnings for
             final_message = set([])
             for idx, msg in enumerate(warning_list):
-                if idx == 0:
-                    final_message = final_message.union(msg.message.components)
-                else:
-                    final_message = final_message.intersection(msg.message.components)
-            warnings.warn(ParameterNotUsedWarning(final_message))
+                if isinstance(msg.message, ParameterNotUsedWarning):
```

Comment:
There might be a bug here? What if the warning list is `[RandomWarning(...), ParameterNotUsedWarning(...)]`.
Then `idx` will be 1 when the `isinstance` check is true so we'll do the intersection of an empty set and the `ParameterNotUsedWarning`, so we'll wipe out the warning we want to log?

Reply:
Oh shoot, I just merged this but you're totally right 🤦‍♀️ 

It should be a pretty simple fix to replace the idx check with checking if the set is empty or not instead - I'll put up a PR for that shortly!

Reply:
Actually, there is a bug here, but it's not that! The check of `if len(warning_list) == len(self.allowed_pipelines) and len(warning_list) > 0:` would prevent the warning list of `[RandomWarning(...), ParameterNotUsedWarning(...)]` from ever reaching that point in the code, since in that case `len(warning_list != len(self.allowed_pipelines)`.

So in this case we would indeed fail to log the `ParameterNotUsedWarning`, but not for the reason highlighted. 

Also, this highlights another issue where we currently catch and then do not raise all non-`ParameterNotUsedWarning` warnings during `make_pipeline`. Some sleuthing found that PR #2564 added the warning catches where they were not before, so this behavior was unwanted but introduced then. I'm going to file an issue and put up a PR for raising all warnings here again, since the size of this change just grew.